### PR TITLE
feat: Convert between Excel file and `Table`

### LINKED ISSUE #138: Create a `Table` from an Excel file
### Is your feature request related to a problem?

Tabular data is often stored in Excel files. We are currently not able to create a `Table` from them directly.

### Desired solution

Add a new method to the `Table` class:

```py
class Table:
    @staticmethod
    def from_excel_file(path: str) -> Table:
       ...
```

### Possible alternatives (optional)

_No response_

### Screenshots (optional)

_No response_

### Additional Context (optional)

If possible, we should use the first row as the column names. Otherwise, we should name the columns `column_0`, `column_1`, etc.

### LINKED ISSUE #139: Write `Table` to an Excel file
### Is your feature request related to a problem?

If we can create a `Table` from an Excel file (#138), we should also be able to create an Excel file for the `Table`.

### Desired solution

A new method in the `Table` class:

```py
class Table:
   def to_excel_file(self, path: str) -> None:
      ...
```

### Possible alternatives (optional)

_No response_

### Screenshots (optional)

_No response_

### Additional Context (optional)

* The column names should be added as the first row.
* If #146 is done before this, `path` should have type `str | Path`.

### PR DESCRIPTION
**Pull request by PhilipGutberlet**
Closes #138. 
Closes #139 .

### Summary of Changes
Created the methods to_excel_file() and from_excel_file() in Class Table.

Co-authored-by: patrikguempel <128832338+patrikguempel@users.noreply.github.com>

### CODE REVIEW

**Comment by lars-reimann** [comment]
## [🦙 MegaLinter](https://megalinter.io/6.22.1) status: ❌ [ERROR](https://github.com/Safe-DS/Stdlib/actions/runs/4765045311)

| Descriptor  |                                 Linter                                 |Files|Fixed|                            Errors                            |Elapsed time|
|-------------|------------------------------------------------------------------------|-----|----:|--------------------------------------------------------------|------------|
|✅ PYTHON    |[black](https://megalinter.io/6.22.1/descriptors/python_black)          |    3|    3|                                                             0|1.31s       |
|✅ PYTHON    |[mypy](https://megalinter.io/6.22.1/descriptors/python_mypy)            |    3|     |                                                             0|2.05s       |
|❌ PYTHON    |[ruff](https://megalinter.io/6.22.1/descriptors/python_ruff)            |    3|    3|[4](https://github.com/Safe-DS/Stdlib/actions/runs/4765045311)|0.05s       |
|✅ REPOSITORY|[git_diff](https://megalinter.io/6.22.1/descriptors/repository_git_diff)|yes  |     |no                                                            |0.03s       |

See detailed report in [MegaLinter reports](https://github.com/Safe-DS/Stdlib/actions/runs/4765045311)
_Set `VALIDATE_ALL_CODEBASE: true` in mega-linter.yml to validate all sources, not only the diff_

_MegaLinter is graciously provided by [OX Security](https://www.ox.security/?ref=megalinter)_

File: src/safeds/data/tabular/containers/_table.py [hunk 1]
```diff
@@ -1141,6 +1168,23 @@ def to_csv_file(self, path: str) -> None:
         data_to_csv.columns = self._schema.column_names
         data_to_csv.to_csv(path, index=False)
 
+    def to_excel_file(self, path: str) -> None:
+        """
+        Write the data from the table into a Excel file.
```
Comment (lars-reimann) [inline src/safeds/data/tabular/containers/_table.py:1173]:
```suggestion
        Write the data from the table into an Excel file.
```

File: tests/safeds/data/tabular/containers/_table/test_from_excel_file.py [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+from safeds.data.tabular.containers import Table
+
+from tests.helpers import resolve_resource_path
+
+
+def test_from_excel_file_valid() -> None:
+    path = "./test.xlsx"
+    table = Table.from_dict({"A": ["1"], "B": ["2"]})
+    tmp_table_file = openpyxl.Workbook()
+    tmp_table_file.save(path)
```
Comment (lars-reimann) [inline tests/safeds/data/tabular/containers/_table/test_from_excel_file.py:14]: The Excel file to read should be uploaded to the repository and just read here.

**Review by lars-reimann — CHANGES_REQUESTED** [review]

File: tests/safeds/data/tabular/containers/_table/test_from_excel_file.py [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+    try:
+        with Path(path).open("w", encoding="utf-8") as _:
+            table.to_excel_file(path)
+        table = Table.from_excel_file(path)
+        assert table.get_column("A").get_value(0) == 1
+        assert table.get_column("B").get_value(0) == 2
```
Comment (Marsmaennchen221) [inline tests/safeds/data/tabular/containers/_table/test_from_excel_file.py:20]:
```suggestion
        table_from_excel = Table.from_excel_file(path)
        assert table_from_excel == table
        assert table_from_excel.get_column("A").get_value(0) == 1
        assert table_from_excel.get_column("B").get_value(0) == 2
```

File: tests/safeds/data/tabular/containers/_table/test_from_excel_file.py
Comment (Marsmaennchen221) [inline tests/safeds/data/tabular/containers/_table/test_from_excel_file.py:1]: You could also add a test where you read from an existing excel file

**Review by Marsmaennchen221 — DISMISSED** [review]

File: tests/safeds/data/tabular/containers/_table/test_from_excel_file.py (hunk 3 above)
Comment (lars-reimann) [reply tests/safeds/data/tabular/containers/_table/test_from_excel_file.py:20]: The additional assertions after `assert table_from_excel == table` are not needed, right?

**Comment by lars-reimann** [comment]
Now that #146 is merged, this PR has to be updated as well, so users can pass a path.

File: pyproject.toml [hunk 4]
```diff
@@ -7,7 +7,7 @@ license = "MIT"
 readme = "docs/README.md"
 repository = "https://github.com/Safe-DS/Stdlib"
 documentation = "https://safe-ds-stdlib.readthedocs.io"
-keywords = ["data-science", "machine-learning", "usability", "learnability"]
```
Comment (lars-reimann) [inline pyproject.toml:10]: Should not be changed.

File: pyproject.toml [hunk 5]
```diff
@@ -20,6 +20,8 @@ pandas = "^2.0.0"
 pillow = "^9.5.0"
 scikit-learn = "^1.2.0"
 seaborn = "^0.12.2"
+polars = {extras = ["pandas", "pyarrow", "xlsx2csv"], version = "^0.17.5"}
```
Comment (lars-reimann) [inline pyproject.toml:23]: Should not be added.

File: poetry.lock
Comment (lars-reimann) [inline poetry.lock:1]: Regenerate the lock file after adjusting `pyproject.toml` with `poetry lock --no-update`.

File: src/safeds/data/tabular/containers/_table.py [hunk 6]
```diff
@@ -1206,6 +1236,44 @@ def to_csv_file(self, path: str | Path) -> None:
...
+        #Create Excel metadata in the file
+        tmp_table_file = openpyxl.Workbook()
+        tmp_table_file.save(path)
+
+        Path(path).parent.mkdir(parents=True, exist_ok=True)
+        data_to_excel = self._data.copy()
+        data_to_excel.columns = self._schema.column_names
+        data_to_excel.to_excel(path)
+
+    def to_excel_file(self, path: str | Path) -> None:
```
Comment (lars-reimann) [inline src/safeds/data/tabular/containers/_table.py:1260]: Why do you have this twice?

File: tests/safeds/data/tabular/containers/_table/test_from_excel_file.py [hunk 7]
```diff
@@ -0,0 +1,26 @@
...
+
+@pytest.mark.parametrize(
+    ("path"),
+    [
+        "./dummy_excel_file.xlsx",
+        Path("./dummy_excel_file.xlsx")
+    ],
+    ids=["string path", "object path"],
+)
+def test_from_excel_file_valid(path) -> None:
```
Comment (lars-reimann) [inline tests/safeds/data/tabular/containers/_table/test_from_excel_file.py:18]: Use the `test_should` naming schema for tests. Same for the other tests.

**Review by lars-reimann — CHANGES_REQUESTED** [review]

**Comment by lars-reimann** [comment]
Please finish this next week before starting something new.

File: tests/safeds/data/tabular/containers/_table/test_from_excel_file.py (hunk 3 above)
Comment (PhilipGutberlet) [reply tests/safeds/data/tabular/containers/_table/test_from_excel_file.py]: This is redundant considering the excel file is now staticly read. (There is no 'table' to compare to)

File: pyproject.toml (hunk 4 above)
Comment (lars-reimann) [reply pyproject.toml:10]: Please read the comments and either write why you don't want to work on it or fix them. Don't just resolve them without doing anything.

**Review by lars-reimann — APPROVED** [review]

**Pull request merged**